src/audio/base_fw.c

```diff
@@ -0,0 +1,286 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <sof/audio/component.h>
+#include <sof/lib/memory.h>
+#include <sof/ut.h>
+#include <ipc4/base_fw.h>
+#include <ipc4/pipeline.h>
+#include <version.h>
+
+/* 0e398c32-5ade-ba4b-93b1-c50432280ee4 */
+DECLARE_SOF_RT_UUID("basefw", basefw_comp_uuid, 0xe398c32, 0x5ade, 0xba4b,
+		    0x93, 0xb1, 0xc5, 0x04, 0x32, 0x28, 0x0e, 0xe4);
+DECLARE_TR_CTX(basefw_comp_tr, SOF_UUID(basefw_comp_uuid), LOG_LEVEL_INFO);
+
+static inline void set_tuple(struct ipc4_tuple *tuple, uint32_t type, uint32_t length, void *data)
+{
+	tuple->type = type;
+	tuple->length = length;
+	memcpy_s(tuple->data, length, data, length);
+}
+
+static inline void set_tuple_uint32(struct ipc4_tuple *tuple, uint32_t type, uint32_t value)
+{
+	tuple->type = type;
+	tuple->length = sizeof(uint32_t);
+	memcpy_s(tuple->data, tuple->length, &value, tuple->length);
+}
+
+static inline struct ipc4_tuple *next_tuple(struct ipc4_tuple *tuple)
+{
+	return (struct ipc4_tuple *)((char *)(tuple) + sizeof(*(tuple)) + tuple->length);
+}
+
+static int basefw_config(uint32_t *data_offset, char *data)
+{
+	uint32_t version[4] = {SOF_MAJOR, SOF_MINOR, SOF_MICRO, SOF_BUILD};
+	struct ipc4_tuple *tuple = (struct ipc4_tuple *)data;
+	struct ipc4_scheduler_config sche_cfg;
+
+	set_tuple(tuple, IPC4_FW_VERSION_FW_CFG, sizeof(version), version);
+
+	tuple = next_tuple(tuple);
+	set_tuple_uint32(tuple, IPC4_MEMORY_RECLAIMED_FW_CFG, 1);
+
```

Comment:
Is this part of the extended manifest ?

Reply:
no, it is part of base fw.

Reply:
the comments for IPC4_MEMORY_RECLAIMED_FW_CFG is "Indicates whether legacy DMA memory is managed by FW". A little confusing.

Reply:
I also checked other items and manifest.  these items are not included by manifest.

Reply:
yes, host query these values upon boot. Some items are read from hw registers, so we can't set them in manifest. We can store these tuples in dynamic memory and free it when get function return since get function is a low-frequency case.

Reply:
ok, this should be in short term.